iterator_test.go

```diff
@@ -2737,3 +2738,179 @@ func BenchmarkSeekPrefixTombstones(b *testing.B) {
 		iter.SeekPrefixGE(seekKey)
 	}
 }
+
+// BenchmarkQueueWorkload benchmarks a workload consisting of multiple queues
+// that are all being processed at the same time. Processing a queue entails either
+// appending to the end of the queue (a Set operation) or deleting from the start of
+// the queue (a Delete operation). The goal is to detect cases where we see a large
+// buildup of point tombstones at the beginning of each queue, which leads to the
+// slowdown of SeekGE(<start of queue>). To that end, the test progresses through
+// a series of steps that each 1) process the queues a certain number of times
+// and then 2) benchmark both the queue processing throughput and SeekGE performance.
+// See https://github.com/facebook/rocksdb/wiki/Implement-Queue-Service-Using-RocksDB
+// for more information.
+func BenchmarkQueueWorkload(b *testing.B) {
+	const queueCount = 25
```

Comment:
A small number of queues than 25 would probably get us bigger clumps of tombstones. You could also vary this in sub-benchmarks, eg.

```
queueCounts := int[]{1,2,4,8,16}
for _, queueCount := range queueCounts {
  b.Run(fmt.Sprintf("queues=%d", queueCount), func(...){
    ....
  })
}
```

Reply:
Lowered this to 8, which I think is working well